src/components/details/ArgoCdLink.tsx

```diff
@@ -0,0 +1,39 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { SplitItem } from '@patternfly/react-core';
+import * as argoIcon from '../../images/argo.png';
+import ExternalLink from '../utils/ExternalLink/ExternalLink';
+
+type ArgoCdLinkProps = {
+  envName: string;
+  appName: string;
+  argocdLink: any;
+};
+
+const ArgoCdLink: React.FC<ArgoCdLinkProps> = ({
+  envName,
+  appName,
+  argocdLink,
+}) => {
+    // Use environment name as is or the original KAM-based design
+    const appNameLink = envName.includes(appName) ? envName : envName + "-" + appName;
```

Comment:
Is this condition to distinguish KAM-based env and non-KAM-based env reliable? If user use KAM to create an app and add env to it, it's totally possible for the user to name the env with the app name as a pre-fix right? Is there another way to tell if an env is KAM-based or not?

Reply:
To support both ways in the interim, we have to set a limitation on the name.  Because kam apps always expects the name of the Argo CD application environment name to be _{envName}-{appName}_, the non-KAM case can either follow this convention, which is reasonable, or  include the appName as part of the environment name, like myapp-dev or myapp-staging.  That way, we won't erroneously append the appName to it, like mapp-dev-myapp.  This latter convention was used by Gerald.

We can implement this perhaps with a new API in the backend or use another label or other mechanism to clearly distinguish a kam generated app from a non-kam app. (I suspect this will not be compatible with older versions of OpenShift)